WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js

```diff
@@ -538,10 +538,20 @@ function() {
 */
 ZmCalendarTreeController.prototype._addExternalCalendarListener =
 function() {
-	var dialog = this.getExternalCalendarDialog();
-    dialog.popup();
+	return AjxRpc.invoke(null, "/service/extension/oauth2/info/google", null, new AjxCallback(this, this._handleGoogleConfig), true);
 };
 
+ZmCalendarTreeController.prototype._handleGoogleConfig =
+function(response){
+	try {
+		var responseJson = JSON.parse(response.text);
+	} catch(e) {
+		console.error("unable to handle Google Config get info response => ", e);
+	}
+
+	var dialog = this.getExternalCalendarDialog(responseJson.data && responseJson.data.client_id);
```

Comment:
better to check for existence of responseJson also
when service returns invalid json at that time responseJson is undefined and accessing responseJson.data will raise javascript error

Reply:
ok will add it for more safety i already tested this on when configured || not configured